python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -503,6 +503,18 @@ def read(self, timeout: Optional[float] = None) -> Any:
 
         return ret
 
+    def release_buffer(self, timeout: Optional[float] = None) -> None:
+        assert (
+            timeout is None or timeout >= 0 or timeout == -1
+        ), "Timeout must be non-negative or -1."
```

Comment:
Should this be moved closer to where the timeout is acquired? In `compiled_dag_node.py:release_output_channel_buffers` perhaps?